enhancements/update/upgrade-without-registry.md

```diff
@@ -0,0 +1,989 @@
+
+      # This is an optional digest of the bundle. If set then the cluster version
+      # operator will ensure that the bundle is used only if it matches this
+      # digest. This is intended to prevent use of a wrong USB stick, or use of
+      # a perfectly fine USB stick in the wrong cluster. The administrator of
+      # the cluster sets it in advance, before the technician arrives in the
+      # cluster site and plugs the USB stick.
+      digest: sha256:....
+
+status:
+  desired:
+    bundle:
+      # Location of the upgrade bundle. This is populated by the monitor, if
+      # enabled, or copied from the `spec.desiredUpdate.bundle.file` field if
+      # monitoring is disabled.
+      file: /root/my-bunde.tar # Can be a file.
+      file: /dev/sdb # Or a device.
+
+      # Node where the bundle is available. This is optional. By default the
+      # cluster version operator will look for the bundle in all the nodes, but
+      # if this is used it will look only in the given node.
+      node: node2
+
+      # Contains the current status of the bundle processing in each node
+      # of the cluster. Keys are the node names.
+      nodes:
+        node0:
+          # Indicates if the upgrade bundle has already been extracted to
+          # the `/var/lib/upgrade/4.13.7-x86_64` directory.
+          extracted: true|false
+          # Indicates if the images from the upgrade bundle have been already
+          # copied to the `/etc/containers/storage` directory.
+          loaded: true|false
+          # Contains the metadata of the bundle as seen by this node. This is
+          # taken from the `metadata.json` file that is part of the bundle.
+          metadata: '{ ... }'
+        node1: { ... }
+        ...
+```
+
+And this is a summary of the new `ContainerRuntimeConfig` fields:
+
+```yaml
+apiVersion: machineconfiguration.openshift.io/v1
+kind: ContainerRuntimeConfig
+metadata:
+  name: pin-upgrade
+spec:
+  containerRuntimeConfig:
+
+    # Additional image store directories. Translates into something like this
+    # inside the `/etc/containers/storage.conf` file:
+    #
+    # additionalimagestores = [
+    #   "/var/lib/my-images/...",
+    #   "/var/lib/your-images/...",
+    #   ...
+    # ]
+    additionalImageStores:
+    - /var/lib/my-images
+    - /var/lib/my-images
+
+    # List of pinned and pre-loaded images. Translates into a new `pin-upgrade.conf`
+    # file inside the `/etc/crio/crio.conf.d` directory with content similar to this:
+    #
+    # pinned_images = [
+    #   "quay.io/openshift-release-dev/ocp-release@sha256:...",
+    #   "quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:...",
+    #   "quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:...",
+    #   ...
+    # ]
+    #
+    # The CRI-O service will be reloaded when this is changed.
+    #
+    # The machine config operator will ensure that these images are pre-loadd and
+    # in all the nodes of the cluster.
+    pinnedImages:
```

Comment:
This doesn't seem to exist today?

Reply:
No, this doesn't exist today, we are suggesting to add it.